### ISSUE TITLE
Issue when upgrading from < 1.3 to >= 1.3 with api keys set to empty string

### ISSUE THREAD

**Issue by mmachatschek** [meilisearch/meilisearch-php#584]
**Description**

PR #514 cleaned up some stuff around the HTTP Client and made some improvements. Unfortunately the PR introduced an breaking change where previously an empty string `''` would not add authorization headers to subsequent api requests to the Meilisearch instance.

e.g. PHP env files at Laravel, this env var would be an empty string:

```
MEILISEARCH_KEY=
```

**Expected behavior**

This code should not add an authorization header to subsequent Meilisearch requests.

```
use Meilisearch\Client;
$meilisearchClient = new Client(
  'http://127.0.0.1:7700',
  '',
);
```

**Current behavior**

The above code adds auth headers to Meilisearch requests.

**Screenshots or Logs**

https://github.com/laravel/scout/issues/769

**Environment (please complete the following information):**
- meilisearch-php version: >= 1.3.0

**Comment by mmachatschek** [comment]
FIxed by #585

**Comment by curquiza** [comment]
v1.4.1 is released, fixing the issue
